code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/interfaces/TokenUtil.java

```diff
@@ -28,7 +28,7 @@
 
     Account getOrCreateAccount(Identity user, Set<Identity> identities, Account account);
 
-    Token createToken(Set<Identity> identities, Account account);
+    Token createToken(Set<Identity> identities, Account account, String originalLogin);
```

Comment:
Can you create a overloaded method here and keep the original unchanged - this way not all auth providers need to change - only those who have this domain\username issue can use the new method.